resources/test_fonts/rebuild.sh

```diff
@@ -0,0 +1,39 @@
+#!/usr/bin/env bash
+
+# run this script from the repo root to rebuild the binary inputs (ttfs) from
+# their xml/ttx representations.
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+SRC_DIR=./resources/test_fonts/ttx
+OUT_DIR=./resources/test_fonts/ttf
+VENV_DIR=./.venv
+PIP=$VENV_DIR/bin/pip
```

Comment:
Establishing the venv already sets pip to point to the right place, what's the point of having this?

Reply:
(similar for ttx)

Reply:
because he has not activated the venv so the venv's bin directory is not in the environment's PATH variable. I suggested Colin to **not** activate the venv but instead invoke the venv scripts (pip, ttx, etc.) directly using their relative paths, it works both ways.

Reply:
yes I call the executables directly, instead of activating the venv. This was per @anthrotype's suggestion, but I can see it being nice in the case where the user has an active venv that I don't want to clobber?

Reply:
It should work either way, because activating a venv from within another venv would place the former at the beginning of the $PATH, and calling activate from within the shell script won't modify the environment of the parent shell that's invoking the script (unless you do source rebuild.sh). I think I prefer the relative path approach without the venv activation, it's more explicit what executable you are actually invoking

Reply:
On the other hand, an argument in favour of doing `source .venv/bin/activate` would be if the python code you are invoking in turn relies on something else being on the $PATH (e.g. it may invoke some other tool as a subprocess and expect that to be on the $PATH and you may want it to invoke the copy which is inside the venv's bin instead of the one from the global environment). But this unfrequent and definitely is not the case for ttx